src/Geometry/CartesianDiscreteModels.jl

```diff
@@ -199,3 +236,132 @@ function _fix_dface_geolabels!(
+    nfaces = length(face_to_geolabel)
+    for face_gid = 1:nfaces
+      cell_gid = face_to_cells.data[face_to_cells.ptrs[face_gid]]
+      a = cell_to_faces.ptrs[cell_gid]
+      b = cell_to_faces.ptrs[cell_gid+1] - 1
+      gci = (subcis[cell_gid] + minus_one_ci) + cmin
+
+      face_lid = -1
+      for j = a:b
+        if (cell_to_faces.data[j] == face_gid)
+          face_lid = j - a + 1
+          break
+        end
+      end
+      @assert face_lid != -1
+
+      face_lid += offsets[d+1]
+      # Check whether cell neighbour across face face_lid belongs to the
+      # global grid. If yes, the current face is actually at the interior
+      is_assigned_face_delta = isassigned(face_deltas, face_lid)
+      if (is_assigned_face_delta && (gci + face_deltas[face_lid]) in gcis)
+        face_to_geolabel[face_gid] = interior_id
+      else
+        cell_found = false
+        for j = d+1:D-1
+          dface_to_jfaces =
+            polytope_d_face_to_jfaces[d+1, j+1][face_lid-offsets[d+1]]
+          cell_found = _is_there_interior_cell_across_higher_dim_faces(
+            dface_to_jfaces,
+            offsets[j+1],
+            gcis,
+            gci,
+            face_deltas,
+          )
+          cell_found && break
+        end
+        if (cell_found)
+          # The current face is at least in two subgrid cells
+          face_to_geolabel[face_gid] = boundary_id
+        else
+          # The current face is only in one subgrid cell
+          face_to_geolabel[face_gid] = face_lid
+        end
+      end
+    end
+  end
+  _fix_geolabels(D, topo, d_to_dface_to_entity, interior_id, boundary_id)
+  fill!(d_to_dface_to_entity[end], interior_id)
+end
+
+function _is_there_interior_cell_across_higher_dim_faces(
+  dface_to_jfaces,
+  offset_j,
+  gcis,
+  gci,
+  face_deltas,
+)
+  for k in dface_to_jfaces
+    jface_lid = k + offset_j
+    if (isassigned(face_deltas, jface_lid))
+      if ((gci + face_deltas[jface_lid]) in gcis)
+        return true
+      end
+    end
+  end
+  return false
+end
+
+"""
+  _find_ncube_face_neighbor_deltas(p::Polytope{D}) -> Vector{CartesianIndex}
+
+  Given an n-cube type Polytope{D}, returns V=Vector{CartesianIndex} with as many
+  entries as the number of faces in the boundary of the Polytope. For an entry face_lid
+  in this vector, V[face_lid] returns what has to be added to the CartesianIndex of a
+  cell in order to obtain the CartesianIndex of the cell neighbour of K across the face F
+  with local ID face_lid.
+"""
+function _find_ncube_face_neighbor_deltas(p::Polytope{D}) where {D}
+  nfaces = num_faces(p)
+  delta_faces = Vector{CartesianIndex}(undef, nfaces - 1)
+  for face_lid = 1:nfaces-1
+    delta_faces[face_lid] = _find_ncube_face_neighbor_delta(p, face_lid)
+  end
+  delta_faces
+end
+
+function _find_ncube_face_neighbor_delta(p::Polytope{D}, face_lid) where {D}
+  @assert is_n_cube(p)
+  result = fill(0, D)
+  face = p.dface.nfaces[face_lid]
```

Comment:
This function is not valid for any `Polytope`, only for `ExtrussionPolytope` (note that you accessing the field `dface`)

Change the function signature to
```julia
function _find_ncube_face_neighbor_delta(p::ExtrussionPolytope{D}, face_lid) where {D}
```

and implement a fallback function
```julia
function _find_ncube_face_neighbor_delta(p::Polytope{D}, face_lid) where {D}
  @notimplemented "Only implemented for ExtrusionPolytope"
end
```

Please check if other parts of the new code are also specific for `ExtrusionPolytope`

Reply:
> This function is not valid for any `Polytope`, only for `ExtrussionPolytope` (note that you are accessing the field `dface`)

Ok. Got it!

> Change the function signature to
```julia
function _find_ncube_face_neighbor_delta(p::ExtrusionPolytope{D}, face_lid) where {D}
```
Done. e4eda3cf9c768438b75cd7ebca348e5076a3ac4d

> and implement a fallback function
```julia
function _find_ncube_face_neighbor_delta(p::Polytope{D}, face_lid) where {D}
  @notimplemented "Only implemented for ExtrusionPolytope"
end
```

I think there is no need to implemented a fallback function. Please note that the functions (now) restricted to `ExtrusionPolytope{D}` I have added are helper functions which are only used in the domain space of `CartesianDiscreteModels.jl`. In other words, these are NOT methods bound to `Polytope{D}`'s API. Agreed?

> Please check if other parts of the new code are also specific for `ExtrusionPolytope`

Did it. The aforementioned two helper functions are the only ones which are specific to ExtrusionPolytope.